apps/whale-api/src/module.api/governance.service.ts

```diff
@@ -51,6 +51,9 @@ export class GovernanceService {
     })
 
     return ApiPagedResponse.of(proposals, size, (proposal) => {
+      if (proposal === undefined) {
+        return undefined
+      }
```

Comment:
When we are trying to get all proposals, we are passing `limit: 0`, and if data happens to be empty, line 90 throws error
https://github.com/JellyfishSDK/jellyfish/blob/e212ed1c5909dc40509fbfe8f970b1e612cd54fa/apps/whale-api/src/module.api/_core/api.paged.response.ts#L88-L95

Reply:
How come it can be undefined in a list?

Reply:
Because you edited `apps/whale-api/src/module.api/_core/api.paged.response.ts`, I'm asking cause when you edit this, you might change the fundamentals of how a list responses can be handled/return changing all implementation contractual expectation.

Reply:
The list is not undefined, the value being accessed by the array index is undefined, i.e.: `data[-1]`.
This will only happen when `limit: 0` and when the list returned is empty. 

Okay, understand where you are coming from now, since my changes will open up the possibility that a valid paged response to have `nextToken: undefined`

Perhaps i shall:
- add this specific condition `if (data.length === limit && data.length > 0 && limit > 0)` explicitly to line 89 in `apps/whale-api/src/module.api/_core/api.paged.response.ts`
- and revert the change i did on line 88 in `apps/whale-api/src/module.api/_core/api.paged.response.ts` ?

Reply:
> Perhaps i shall:

Yeah this is better, but also:

Currently there is also a `@Min(1)` where only `size > 0` would be allowed.

https://github.com/JellyfishSDK/jellyfish/blob/bee44d910bf817e3c41d0117b0cca60b1d86dd55/apps/whale-api/src/module.api/_core/api.query.ts#L15

Would this case be valid in the first place?

Reply:
> Would this case be valid in the first place?

Yea, i didn't want to change the `size` behaviour at controller level, so i added an `all` flag and pass `limit: 0` at the service level.

> Yeah this is better

Will update